### DISCUSSION TITLE
Dielectric loss for wideband models

### DISCUSSION THREAD

**Discussion by VolkerMuehlhaus** [thliebig/openEMS-Project#99] [Q&A Usage]
Hi all,

I would like to follow up on the loss tangent model topic that came up inthe SMA-to-CPW model discussion:
https://github.com/thliebig/openEMS-Project/discussions/88#discussioncomment-6373210

My understanding is that openEMS does not support dielectric loss tangent, and we need to convert loss tangent into conductivity (sigma in the equation below, openEMS parameter Kappa). However, that calculation is valid (exact) at one frequency only.

![tand](https://github.com/thliebig/openEMS-Project/assets/47922981/3098f6e8-e0c5-43ee-ae0c-62aca7d96c23)

To check the effect, I used Thorstens lossy microstrip example and added the tand->kappa calculation at center frequency. Metal is set to almost perfect conductor in that model (very high conductivity), so that we only see the effect of dielectric loss. 

```
substrate.epr = 9.8;
substrate.tand = 0.01
% calculate kappa from loss tangent at center frequency
substrate.kappa = substrate.tand*substrate.epr*EPS0*2*pi*(f_stop-f_start)/2;
(...)
CSX = SetMaterialProperty( CSX, 'RO4350B', 'Epsilon', substrate.epr, 'Kappa', substrate.kappa );
````

Below is the comparison to the Sonnet MoM solver which can handle dielectric loss: the openEMS results agrees at center frequency where the tand->kappa calculation was performed:  ~ 0.28dB insertion loss @ 12.5 GHz.
But wideband, result's don't agree at all: openEMS show almost constant loss from DC to 25 GHz, whereas the Sonnet results scales with frequency as expected.

This is not an issue if conductor loss is much larger than dielectric loss anyway, e.g. for expensive teflon or ceramic substrates, but for other substrates like FR4 dielectric loss tangent is the major loss factor at microwave frequencies. 

For wideband models, we then have an issue, or do I miss something? I'm asking because in the other thread, advanced topics like surface roughness and dispersion were discussed, but don't we have a much more fundamental limitation here?

Best regards
Volker

![tand_testcase_results](https://github.com/thliebig/openEMS-Project/assets/47922981/54ffb7b8-eee7-43f7-a44b-d23022175207)

```%
% examples / microstrip / MSL_Losses
%
% This example demonstrates how to model sheet conductor losses
%
% Tested with
%  - Matlab 2013a / Octave 3.8.1+
%  - openEMS v0.0.32
%
% (C) 2012-2014 Thorsten Liebig <thorsten.liebig@gmx.de>

%
% modification by VM: add loss tangent to dielectric, set metal loss to zero for this tand testcase


close all
clear
clc
confirm_recursive_rmdir(0);   % delete old data without asking

basename = mfilename ; % get name of current model from *.m filename

%% setup the simulation %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
physical_constants;
unit = 1e-6; % specify everything in um

f_start = 0e9;
f_stop  = 25e9;

lambda = c0/f_stop;

MSL.length = 10000;
MSL.port_dist = 5000;
MSL.width = 225;
MSL.conductivity = 9e9;  % conductor almost lossless
MSL.thickness = 35e-6;

substrate.thickness = 250;
substrate.epr = 9.8;
substrate.tand = 0.01
% calculate kappa from loss tangent at center frequency
substrate.kappa = substrate.tand*substrate.epr*EPS0*2*pi*(f_stop-f_start)/2;


%% setup FDTD parameters & excitation function %%%%%%%%%%%%%%%%%%%%%%%%%%%%
FDTD = InitFDTD('endCriteria',1e-4);
FDTD = SetGaussExcite(FDTD,0.5*(f_start+f_stop),0.5*(f_stop-f_start));
BC   = {'PML_8' 'PML_8' 'PML_8' 'PML_8' 'PEC' 'PML_8'};
FDTD = SetBoundaryCond( FDTD, BC );

%% setup CSXCAD geometry & mesh %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
CSX = InitCSX();
resolution = c0/(f_stop*sqrt(substrate.epr))/unit /20;
mesh.x = SmoothMeshLines( [-MSL.length*0.5-MSL.port_dist 0 MSL.length*0.5+MSL.port_dist], resolution, 1.3 ,0 );
mesh.y = SmoothMeshLines2( [0 MSL.width/2], resolution/6 , 1.3);
mesh.y = SmoothMeshLines( [-0.5*lambda/unit -mesh.y mesh.y 0.5*lambda/unit], resolution, 1.4);
mesh.z = SmoothMeshLines( [linspace(0,substrate.thickness,10) 0.5*lambda/unit], resolution );
CSX = DefineRectGrid( CSX, unit, mesh );

%% substrate
CSX = AddMaterial( CSX, 'RO4350B' );
CSX = SetMaterialProperty( CSX, 'RO4350B', 'Epsilon', substrate.epr, 'Kappa', substrate.kappa );
start = [mesh.x(1),   mesh.y(1),   0];
stop  = [mesh.x(end), mesh.y(end), substrate.thickness];
CSX = AddBox( CSX, 'RO4350B', 0, start, stop );

%% MSL ports and lossy line
CSX = AddConductingSheet( CSX, 'gold', MSL.conductivity, MSL.thickness );
portstart = [ mesh.x(1),               -MSL.width/2, substrate.thickness];
portstop  = [ mesh.x(1)+MSL.port_dist,  MSL.width/2, 0];
[CSX, port{1}] = AddMSLPort( CSX, 999, 1, 'gold', portstart, portstop, 0, [0 0 -1], 'ExcitePort', true, 'FeedShift', 10*resolution, 'MeasPlaneShift',  MSL.port_dist);

portstart = [mesh.x(end),              -MSL.width/2, substrate.thickness];
portstop  = [mesh.x(end)-MSL.port_dist, MSL.width/2, 0];
[CSX, port{2}] = AddMSLPort( CSX, 999, 2, 'gold', portstart, portstop, 0, [0 0 -1], 'MeasPlaneShift',  MSL.port_dist );

start = [mesh.x(1)+MSL.port_dist,   -MSL.width/2, substrate.thickness];
stop  = [mesh.x(end)-MSL.port_dist,  MSL.width/2, substrate.thickness];
CSX = AddBox(CSX,'gold',500,start,stop);

%% write/show/run the openEMS compatible xml-file
Sim_Path = ['data/' basename];
Sim_CSX = [basename '.xml'];

[status, message, messageid] = rmdir( Sim_Path, 's' ); % clear previous directory
[status, message, messageid] = mkdir( Sim_Path ); % create empty simulation folder

WriteOpenEMS( [Sim_Path '/' Sim_CSX], FDTD, CSX );
CSXGeomPlot( [Sim_Path '/' Sim_CSX] );
RunOpenEMS( Sim_Path, Sim_CSX ,'');

%% post-processing
close all
f = linspace( f_start, f_stop, 1601 );
port = calcPort(port, Sim_Path, f, 'RefImpedance', 50);

s11 = port{1}.uf.ref./ port{1}.uf.inc;
s21 = port{2}.uf.ref./ port{1}.uf.inc;

plot(f/1e9,20*log10(abs(s21)),'r--','LineWidth',2);
grid on;
hold on;
ylabel('|S_21| (dB)','Interpreter','None');
xlabel('frequency (GHz)');
```

**Comment by thliebig** [comment]
Well as I have stated many times, using just the dielectric conductivity can always only be a narrow band model for the tanD values.
But openEMS has full support for any size Drude/Lorentz models and in very early days there were a calculator to find a matching parameter set for these models to fit to a material with constant tanD. But I think this got lost over time? I would need to check for that. But in any case it would be worth investigating to maybe create a (new) tool for this purpose...

**Reply by VolkerMuehlhaus** [reply]
Thanks Thorsten! 

It would be great if someone has any example, or even better some code, to calculate Drude/Lorentz model parameters from dielectric loss tangent and apply that in openEMS.

**Comment by toammann** [comment]
Hello,
@biergaizi suggested [here](https://github.com/thliebig/openEMS-Project/discussions/88#discussioncomment-6321272) some time ago to look into the Djordjevic-Sarkar model. He was right with his tip to look into the Djordjevic-Sarkar model. Thank you very much for pointing me in this direction! 

I finally found some time to take a deeper look. The Djordjevic-Sarkar model [1] was originally developed for FR4-type materials based on measured data.  The idea is to have epsilon'' (imaginary part) approximately constant between two conrner frequencies w1 = 10^m1 (lower limit) and w2 = 10^m2 (upper limit). With this assumption it is possible to create a wideband model with only one epsilonR, tand pair at a single frequeny (I think this is what many users want and are used to from commercial software). According to multiple sources this model works well for materials with an almost constant loss tangend (very common for PCB materials).

I think it would be a valuable option to have such a simple model in openEMS. The Djordjevic model is roughly speaking a Debye model with an infinite number of terms, but it can be approximated by a finite number of poles. With some math it is possible to solve for the unknowns by choosing an appropriate number of Terms. In a second step the Debye model can be (approximately) converted into the openEMS Lorentz model as defined in openEMS ([details here](https://wiki.openems.de/index.php/Dispersive_Material_Property.html)).

The current state of my work is an octave script that takes a epsilonR and a tand value at one frequency and outputs the values for the openEMS Lorentz terms for use in a simulation script.

The example from above for Er=9.8, tand=0.01, f_fit=1GHz and 6 Terms:
![model_fit_er9 8](https://github.com/thliebig/openEMS-Project/assets/9382233/26b6f2c9-8fb1-4aa0-9c8e-8d8f7261ae39)

By using more terms the fit can be extended to lower frequencies.

The multi-term Debye model "wrapps" around the idealized Djordjevic-Sarkar model. The openEMS Lorentz model follows the Debye model closely with my chosen set of parmeters.

In openEMS the 6-Term material model is defined in the following way:
```
CSX = AddLorentzMaterial(CSX, 'LorentzMat');
CSX = SetMaterialProperty(CSX,'LorentzMat','Epsilon', 9.5757, 'Kappa', 0); % Epsilon is here EpsilonInf
CSX = SetMaterialProperty(CSX,'LorentzMat','EpsilonPlasmaFrequency',   420.2395e+09, 'EpsilonLorPoleFrequency',   3.1623e+012, 'EpsilonRelaxTime',   15.9155e-21);
CSX = SetMaterialProperty(CSX,'LorentzMat','EpsilonPlasmaFrequency_2', 375.4035e+09, 'EpsilonLorPoleFrequency_2', 3.1623e+012, 'EpsilonRelaxTime_2', 159.1549e-21);
CSX = SetMaterialProperty(CSX,'LorentzMat','EpsilonPlasmaFrequency_3', 380.1256e+09, 'EpsilonLorPoleFrequency_3', 3.1623e+012, 'EpsilonRelaxTime_3', 1.5915e-18);
CSX = SetMaterialProperty(CSX,'LorentzMat','EpsilonPlasmaFrequency_4', 380.1256e+09, 'EpsilonLorPoleFrequency_4', 3.1623e+012, 'EpsilonRelaxTime_4', 15.9155e-18);
CSX = SetMaterialProperty(CSX,'LorentzMat','EpsilonPlasmaFrequency_5', 375.4035e+09, 'EpsilonLorPoleFrequency_5', 3.1623e+012, 'EpsilonRelaxTime_5', 159.1549e-18);
CSX = SetMaterialProperty(CSX,'LorentzMat','EpsilonPlasmaFrequency_6', 420.2395e+09, 'EpsilonLorPoleFrequency_6', 3.1623e+012, 'EpsilonRelaxTime_6', 1.5915e-15);
```

Finally the simulation result of the example suggested by @VolkerMuehlhaus:

![grafik](https://github.com/thliebig/openEMS-Project/assets/9382233/2e38314d-dede-42f5-9d0a-ce22061d15f9)

The result looks similar to the Sonnet simulation from above. It is now also clear to me, that the method of "calculating kappa from tand" is only a very! narrowband approximation. The Kappa/Sigma/Conductiviy in the lorentz-Debeye model is really the DC-conductivity of the substrate which has an influence for low frequencies around DC.

I hope that this is of interest for some pepole. It should also be possible to fit the Debye model to a set of input data points (one term per data point) for materials that are not approximated by the Djordjevic-Sarkar model closely. Is this of interest? It can be found in commercial software.

I will make the equations and documentation available at some time (when I find the time to document it). 

If this is useful: How to make it available for users? As a standalone tool? Or as a function within openEMS? Other suggestions?

Regards,
Tobias Ammann

-------------

[1] Djordjevic, Antonije R., et al. "Wideband frequency-domain characterization of FR-4 and time-domain causality." IEEE Transactions on electromagnetic compatibility 43.4 (2001): 662-667.

**Reply by VolkerMuehlhaus** [reply]
Hi Tobias,

many thanks for looking into this, and providing a possible solution!

I have a question: the result that you show has quite strong variation in er_r' (real part) over frequency. Is this an artefact from your fit, or is this physically correct if we enforce constant loss tangent?

Best regards
Volker

**Reply by toammann** [reply]
Hello Volker,
the "strong looking" variation is due to the logarithmic frequency axis. A very high upper frequency limit of 10 000GHz is plotted to show the decay towards f->infinity. Some tweaking of the pole positions still must be done. But this should be an easy task.

Plotted in linear scale the familiar slowly decreasing epsilonR shape becomes visible. Here is an example for an epsilonR=4.3 fit at 10GHz:

![model_fit_er4 3_lin](https://github.com/thliebig/openEMS-Project/assets/9382233/f781b30a-c5cc-403e-ad4c-5b9f6fa0e92c)

according to the Djordjevic-Sarkar paper is is not correct to have an constant value of tand. He claims that this violates causality. Therefore, the paper suggests an (almost) constant imagniary part of the permittivity over a specified frequency range (flat top in the epsilon'' plot in logarithmic scale). Due to the little variation in the real part tand does not vary by a huge amount,

Regards 
Tobias

**Reply by KJ7LNW** [reply]
@toammann , @thliebig , Thank you for your feedback on this subject.

I, too, am also looking for a way to use the manufacturer's PCB substrate specifications. While it would be neat to see the Djordjevic-Sarkar model implemented in openEMS, is @VolkerMuehlhaus's approximation (pasted below) close enough to be correct for low bandwidth microstrip antennas?

```octave
substrate.epr = 9.8;
substrate.tand = 0.01;
% calculate kappa from loss tangent at center frequency
substrate.kappa = substrate.tand*substrate.epr*EPS0*2*pi*(f_stop-f_start)/2;
```

Relatedly, is there any reason that the microstrip antenna Tutorials contain a kappa calculation, but that the microstrip notch filter does not, as shown below? 

```py

## For antennas, kappa appears to be estimated 
## only at the antenna center frequency:

# from Bent_Patch_Antenna.py
substrate_kappa  = 1e-3 * 2*pi*2.45e9 * EPS0*substrate_epsR
substrate = CSX.AddMaterial('substrate', epsilon=substrate_epsR, kappa=substrate_kappa  )

# from Simple_Patch_Antenna.py
substrate_kappa  = 1e-3 * 2*pi*2.45e9 * EPS0*substrate_epsR
substrate = CSX.AddMaterial( 'substrate', epsilon=substrate_epsR, kappa=substrate_kappa)

## and MSL_NotchFilter.py does not define kappa at all.  Should it?
substrate = CSX.AddMaterial( 'RO4350B', epsilon=substrate_epr)

```

**Comment by KJ7LNW** [comment]
One more question: 

Does the _omega_ value of _kappa_ (2\*pi\*Hz) use our target frequency for simulation, or the measurement frequency from the data sheet?

Laminate manufacturers tend to specify tan-d as measured at a particular frequency(s). For example, [this Rogers data sheet](https://www.rogerscorp.com/-/media/project/rogerscorp/documents/advanced-electronics-solutions/english/data-sheets/ro4000-laminates-ro4003c-and-ro4350b---data-sheet.pdf) shows that they were measured at 10GHz and 2.5GHz on the third row:

![rogers datasheet tand](https://github.com/thliebig/openEMS-Project/assets/93454819/da903a72-ad7e-49f5-9de2-54094b4077b5)

Thus, for RO4003C, if our target simulation frequency is 1GHz, then should we use `0.0027*pi*1e9*Er`, or the data sheet value 2.5GHz as `0.0021*pi*2.5e9*Er`?

**Reply by VolkerMuehlhaus** [reply]
> Does the omega value of kappa (2*pi*Hz) use our target frequency for simulation, or the measurement frequency from the data sheet?

You need to specify your simulation target frequency.

**Reply by toammann** [reply]
Hello,
I totally agree with Volker. To make a decision if the [normal material model](https://wiki.openems.de/index.php/Material_Property.html)  is sufficiently accurate for your simulation  you can plot the tand in Octave. 

This is a plot for your rogers example calculating kappa with tand=0.0021 at 1GHz (assuming that it does not change significantly from 1GHz to 2.5GHz):

![untitled](https://github.com/thliebig/openEMS-Project/assets/9382233/2f0e17c9-def3-43a2-81fb-025d7fb4a46a)

You can clearly see, that tand will fit the 0.0021 value perfectly but only at one specific frequeny. This is your simulation target frequency. So, if your model is narrowband this material model will work just fine for you.  No need to go through the trouble of a wideband dielectric model in this case.  It really depends on the bandwidth of your microstrip antenna.  The absolute frequency range where the simple model fits scales with frequency. 

We can also see, that the tand value goes towards zero for higher frequencies. I think that this is the reason for the "flat" frequency response of the initial openEMS simulation from Volker.

Regards
Tobias 

PS: Here is a script to plot the permittivity model.
```
clear;
close all;

eps0 = 8.8541878e-12;
epsR = 3.38;
tandEval = 0.0021;

f = linspace(0, 20e9, 201);
fEval = 1e9;

fEvalidx = find(f==fEval);
kappa = 2*pi*fEval*eps0*epsR*tandEval;
eps = eps0*epsR - 1i*kappa./(2*pi*f);
tand = -imag(eps)./real(eps);

figure;
plot(f/1e9, real(eps));
title('Permittivity \epsilon'' (real part)');
xlabel('Frequency / GHz');
ylabel('Dimensionless');
grid on;

figure;
plot(f/1e9, tand); hold on;
plot(f(fEvalidx)/1e9, tand(fEvalidx), 'ro');
plot([f(1), f(end)]/1e9, [tandEval, tandEval], '--bk');
title('Loss Tangent / tan_\delta');
xlabel('Frequency / GHz');
ylabel('Dimensionless');
grid on
ylim([0, tandEval*2]);
legend('tan_\delta');
```

**Reply by KJ7LNW** [reply]
Thank you for sharing the plot (and the code for the plot)! Out of curiosity I plotted 2.5GHz w/ the Rogers 0.0021 and 10GHz at 0.0027 using your code (unfortunately I do not know how to put them on the same graph, as I am new to Matlab).  

Can you help me interpret these graphs to understand what they mean with respect to using the manufacturer's tand value at a different frequency? 
- Does this indicate at the tand value is very sensitive to change at lower frequencies? 
- As an example, lets say that I needed a wide bandwidth of 100MHz (in reality, we need 4 MHz). Given the graphs, how could you tell if tand is sufficiently accurate for the antenna model?
- Other observations?

The first graph is the same as yours with a plot trace/key name change. The second two graphs plot the frequencies and tand values provided by the manufacturer.

![image](https://github.com/thliebig/openEMS-Project/assets/93454819/bbf2ce90-9166-44ef-9648-bae745fdfada)

![image](https://github.com/thliebig/openEMS-Project/assets/93454819/7f443f80-428c-441e-bf93-8379fb60444d)

![image](https://github.com/thliebig/openEMS-Project/assets/93454819/8ce3c659-5945-4c7b-a1d6-bc4f04a299e4)

**Reply by VolkerMuehlhaus** [reply]
> Does this indicate at the tand value is very sensitive to change at lower frequencies?

No, this is an artefact from the tand-> sigma calculation used here. What the equation does: it converts the dielectric loss tangent into a conductivity value, turning your dielectric into a weakly conducting material. So you get a resistive material, not a lossy isolator, and that is obviously a big difference at DC. The tand-> sigma math is only valid at that exact frequency only!

The small change in the manufacturer data between 2.5 and 10 GHz could be included in your case by guesstimated extrapolation to 1 GHz, maybe 0.0018, and then use the equation above for openEMS.

**Reply by KJ7LNW** [reply]
Awesome, thanks.

**Reply by VolkerMuehlhaus** [reply]
I haven't answered this part of your question yet:

> As an example, lets say that I needed a wide bandwidth of 100MHz (in reality, we need 4 MHz). Given the graphs, how could you tell if tand is sufficiently accurate for the antenna model?

Let's discuss an example with a bandwidth of +/-20%. The "real" tand is almost constant in that frequency range. 
Using the openEMS approximation of converting substrate loss tangent to substrate conductivity

`substrate.kappa = substrate.tand * substrate.epr * EPS0 * 2*pi*ftarget;`

we get the correct substrate loss at our target frequency, 
overestimate substrate loss by 20% at ftarget-20%
and underestimate substrate loss by 20% at ftarget+20%

**Reply by toammann** [reply]
In addition here is a plot of Volkers example:

![untitled](https://github.com/thliebig/openEMS-Project/assets/9382233/55e77aca-8b61-4141-9c3f-3a815ddb4201)

It`s up to you if this variation of tand within the operating bandwith of your antenna is acceptable or not :)

Regards,
Tobias

**Reply by KJ7LNW** [reply]
thanks. great info!

**Reply by KJ7LNW** [reply]
I wonder if openEMS could (someday) take kappa as a function of Hz. Maybe that is what the Djordjevic-Sarkar model gets at.

**Reply by biergaizi** [reply]
> I wonder if openEMS could (someday) take kappa as a function of Hz. 

You confused cause and effect. We need the Djordjevic-Sarkar model exactly because it allows us to model kappa as a function of Hz in openEMS, which is otherwise difficult if not impossible.

You've obviously missed our prior discussion, so allow me to re-explain everything here: the problem has two parts, the first part is modeling frequency-dependent effects in FDTD, the next part is obtaining the kappa vs. frequency curve without access to the "actual" curve.

1. **Modeling frequency-dependent kappa in FDTD**: Modeling frequency-dependent effects can be relatively simple in frequency-domain methods like BEM or FEM. You can simply use the entire "kappa vs. frequency" curve as the input, and run simulations separately at each frequency. In case of a narrow-band design it's even simpler regardless of your simulation method - one can just select a single value of kappa suitable for your purpose. But when you need both wideband modeling and FDTD, the situation is harder - FDTD is a time-domain method, so each simulation must include the effects of all possible values of kappa across a wide range of frequencies simultaneously.

    1.1. **Standard FDTD's solution:** In FDTD, this is a well-known problem with a well-known solution, any good textbook would tell you that in this case, one should model materials with frequency-dependent kappa using special physics models, including Drude material, Lorentz material and Debye materials. Once the kappa versus frequency curve of the real-world material is measured, they can be curve-fitted into one of these models to allow simulation.  (Note that these models are originally developed in the field of solid-state physics, FDTD only uses them as a calculation device, so these parameters are just curve-fitting parameters and are usually not physically meaningful). 

    1.2. **openEMS's feature:** openEMS already supports Drude and Lorentz materials, so it's already capable of model variable kappa as long as you have the model parameters. But currently it doesn't have any routine to run the curve-fitting process, so an official Octave or Python routine in the future would be useful indeed.

2. **The difficulties of obtaining the actual kappa vs. frequency curve**: The "actual" kappa of dielectric materials are both frequency-dependent and are in complex number. The raw data is often *not* published and not accessible to most circuit board designers. The "kappa" values reported in datasheets are incomplete or are "convenient lies". Sometimes it reports a value at a single point, for example, FR-4 manufacturers only report it at 1 GHz because most designs are centered around this frequency. Sometimes the given kappa values are mathematical fits, rather than a physically meaningful value:

    >  It has been shown in many papers that 2D field solvers predict impedance that agrees within the measurement accuracy of the test equipment used to measure impedance. It has also been shown that equations commonly predict the wrong impedance. Because of this equation error problem, the users of equations have often resorted to iterative, trial-and-error methods for adjusting the equations to allow them to predict the correct impedance. Some PCB fabricators have gotten quite good at “fudging” the values of er used in their calculations in order to get good answers from the equations. The problem with this “fudging” is it requires considerable trial and error with new materials until they are “dialed in.”
    > 
    > Even if there is time to “dial in” a new material by manipulating the er values to arrive at accurate impedance calculations, it leaves the PCB engineer not knowing the true er of the material. This is needed in order to accurately calculate velocity for the purpose of predicting the time of flight along traces.
    >
    > Even with a good impedance predicting tool, such as a 2D field solver, impedances are still often calculated wrong. This is because the er value given for a particular laminate may be incorrect. The most common published values for er are measured at 1 MHz. Figure 16.2 shows that the relative dielectric constant of all common PCB materials decreases with frequency. It also shows that the relative dielectric constant varies with the ratio of glass to resin in the material. Therefore, getting the impedance calculation correct requires using a good tool; knowing the glass to resin ratio of the laminate and knowing the frequency at which the transmission line will be used. Once all three of these factors have been taken into account, the results are predictable and repeatable.
    > 
    > - *Right the First Time, a Practical Handbook on High-Speed PCB and System Design*, by Lee W. Ritchey

5. **Djordjevic-Sarkar model**: To overcome the problem of incomplete kappa, the Djordjevic-Sarkar model was created in the industry several years ago in the context of modeling FR-4 circuit boards used by everyday engineers. Without such a model, one would need to measure a real material sample, and while I'm sure big manufacturers often do this in their R&D labs, but this process requires a wideband microwave VNA and special text fixtures, which are often not available or time-consuming. Djordjevic-Sarkar allows us to reconstruct the full "kappa vs. frequency" curve using just a few data point, as few as only one data point, and it outputs an estimated version of the full curve. It would then be possible to numerically fit the reconstructed curve to Drude/Lorentz/Debye material, just like the case when you have the actual curve via measurements.

This is the route of investigation that @toammann has followed so far, with promising results. He has successfully replicated the Djordjevic-Sarkar paper, developed a numerical optimizer to create Lorenz material parameters for use in openEMS, and performed actual experiments. The latest attempt was reported here: [*Some Simulation Results with Wideband Dielectric models*](https://github.com/thliebig/openEMS-Project/discussions/157).

If I read this report correctly, preliminary results are:

1. Wideband dielectric loss is successfully modeled, both in terms of Djordjevic-Sarkar model itself, and in openEMS simulations, the results agree with SONNET.
2. There are numerical instabilities, as Djordjevic-Sarkar model is designed to use with Debye material model, but openEMS uses Lorentz materials, and fitting Debye's parameters into Lorentz is an imperfect process. In the future, we probably need a native Debye model as well.
3. Wideband conductor loss is not modeled correctly, as the Djordjevic-Sarkar model is only a dielectric material model. One still needs to develop a realistic conductor loss model including the effects of surface roughness, future works are needed to solve this problem to enable very wideband circuit board modeling in openEMS.

**Reply by KJ7LNW** [reply]
> openEMS's feature: openEMS already supports Drude and Lorentz materials, so it's already capable of model variable kappa as long as you have the model parameters. 

Neat, thank you for pointing that out, it allowed me to find this page:  https://wiki.openems.de/index.php/Drude/Lorentz_Material_Property.html

> Djordjevic-Sarkar allows us to reconstruct the full "kappa vs. frequency" curve using just a few data point, as few as only one data point, and it outputs an estimated version of the full curve. It would then be possible to numerically fit the reconstructed curve to Drude/Lorentz/Debye material, just like the case when you have the actual curve via measurements.

amazing.
